Fix text_to_word_sequence with unicode text

nicolewhite opened the pull request:
Fixes #7624

fchollet commented:
Please add a unit test (a test that would have failed with the previous version and that passes with this fix).

nicolewhite commented:
The unicode test I added in `test_text.py` is such a test.

```
(keras-dev) ➜  keras git:(master) ipython
Python 2.7.10 (default, Feb  7 2017, 00:08:15) 
In [1]: from keras.preprocessing.text import text_to_word_sequence
In [2]: text_to_word_sequence(u'hé llo')
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-2-7a5ba6c36430> in <module>()
----> 1 text_to_word_sequence(u'hé llo')

/Users/nicole/GitHub/keras/keras/preprocessing/text.py in text_to_word_sequence(text, filters, lower, split)
     39     if lower:
     40         text = text.lower()
---> 41     text = text.translate(maketrans(filters, split * len(filters)))
     42     seq = text.split(split)
     43     return [i for i in seq if i]

TypeError: character mapping must return integer, None or unicode
```

```
(keras-dev) ➜  keras git:(master) git checkout tokenizer-unicode 
Switched to branch 'tokenizer-unicode'
(keras-dev) ➜  keras git:(tokenizer-unicode) ipython
Python 2.7.10 (default, Feb  7 2017, 00:08:15) 
In [1]: from keras.preprocessing.text import text_to_word_sequence
In [2]: text_to_word_sequence(u'hé llo')
Out[2]: [u'h\xe9', u'llo']
```

nicolewhite commented:
I added another test for the `Tokenizer` class with the same input from #7624 just to be extra thorough.
